src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Repository/OpenReferralDbContextExtension.cs

```diff
@@ -0,0 +1,259 @@
+using FamilyHubs.SharedKernel.OpenReferral.Entities;
+using Microsoft.EntityFrameworkCore;
+using Attribute = FamilyHubs.SharedKernel.OpenReferral.Entities.Attribute;
+
+namespace FamilyHubs.SharedKernel.OpenReferral.Repository;
+
+public class OpenReferralDbContextExtension
+{
+    public void OnModelCreating(ModelBuilder modelBuilder)
```

Comment:
Possibly break this method into small methods

Reply:
> possibly use new(0) instead of new() around list initialisations.

I wouldn't have thought there'd be a different between `new()` and `new(0)` 😮 won't both create empty collections? Is it a memory allocation thing (allocating a collection of size 0 is more efficient than not specifying the size something?)

Reply:
private const int DefaultCapacity = 4; in the implementation of List<T>

Reply:
For the capacity thing, the docs here https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.capacity?view=net-7.0#examples has an example and states:

"The parameterless constructor is used to create a list of strings with a capacity of 0, and the Capacity property is displayed to demonstrate this."

So looks like new() defaults internally as new(0)

Reply:
For the breaking down, I'd rather keep it in its current form for now, reflects the styling of stuff like the Report API that way

Reply:
Just looked at the reflected List<T> implementation and it does default to zero now. Used to be 4. You learn something new every day :-)